README.md

```diff
@@ -150,6 +150,26 @@ Alternatively, curl can be used to send a request to the end point:
 curl -i --header "Content-Type: application/json" --request POST --data '{ "email": "test@email.com" }' http://localhost:3002/register
 ```
 
+# Dependency management
+
+Dependencies should be managed within a container using the development image for the app. This will ensure that any packages with environment-specific variants are installed with the correct variant for the contained environment, rather than the host system which may differ between development and production.
+
+The [`cmd`](./scripts/cmd) script is provided to run arbitrary commands in a container using the development image.
+
+Since dependencies are installed into the container image, a full build should always be run immediately after any dependency change.
+
+In development, the `node_modules` folder is mounted to a named volume. This volume must be removed in order for changes to `node_modules` to propagate from the rebuilt image into future instances of the app container. The [`reset`](./scripts/stop) script will rebuild images and reset local containers and volumes using the updated images.
+
+The following example will update all dependencies.
+
+```
+# Run the NPM update
+scripts/cmd npm update
+
+# Remove existing volumes and restart the service with updated images
+scripts/reset
+```
+

```

Comment:
I think is good at explaining the issue with the node modules mount, but it would be good if it also explained the issues that could happen when running tests within a container too.  Just so someone who has issues running tests because of a cached volume would know they can easily solve it with the reset script.

Reply:
Given that the test script already cleared out the volumes after a successful test, I've made it also clear volumes before the tests. That should prevent the cached volume issue for tests in future.